core/src/main/java/org/polypheny/db/algebra/AlgHomogeneousShuttle.java

```diff
@@ -68,7 +68,7 @@ public AlgNode visit( LogicalMatch match ) {
 
 
     @Override
-    public AlgNode visit( Scan scan ) {
+    public AlgNode visit( RelScan<?> scan ) {
```

Comment:
Why relational in the generic alg classes?

Reply:
Adjusted visitor to now correctly use all the concrete model classes.